Meziantou.Xunit.ParallelTestFramework.Tests/ConcurrencyFixture.cs

```diff
@@ -0,0 +1,19 @@
+namespace Meziantou.Xunit.ParallelTestFramework.Tests;
+
+public class ConcurrencyFixture
+{
+    private int concurrency;
+
+    public async Task<int> CheckConcurrencyAsync()

```

Comment:
Tests based on delay are subject to flakiness. Also, I'm not sure `var overlap = concurrency` is safe. I wonder if you need to use `Volatile.Read` (I'm not an expert).

I would use 2 different methods:
- CheckConcurrency
- CheckNoConcurrency

````
// pseudo-code, not tested, not written in VS!
int expectedConcurrency = 2;
TaskCompletionSource tcs = new();
public async Task CheckConcurrency()
{
    if(Interlocked.Decrement(ref expectedConcurrency ) == 0)
    {
        // The last test is running concurrently
        tcs.TrySetResult();
    }
    else
    {
        // wait for other tests to run
        await tcs.Task;
    }
}
````

````
// pseudo-code, not tested, not written in VS!
SemaphoreSlim semaphore = new(1,1);
public async Task CheckNoConcurrency()
{
    Assert.True(semaphore.Wait(0));
    // Wait for other tests to potentially run
    // I don't think we can do better
    await Task.Delay(1000);
    semaphore.Release();
}
````

Reply:
I definitely agree that it's not ideal. I did something similar to your suggestions at first, but realized that both approaches require some level of timing (`CheckConcurrency` would also require a timeout in case it's only called once).

I went with this approach for simplicity and flexibility in case there ever needs to be a collection with more than two tests in the future. The 100 millisecond timeout has been reliable for me so far, but I can change it if you're not comfortable with it.

Reply:
It seems ok on the CI. Let's keep it this way and fix it if flakiness occurs.